main.go

```diff
@@ -1,133 +1,151 @@
-package main
-
-import (
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"log"
-	"net/http"
-	"sync"
-	"strconv"
-
-	"github.com/gorilla/mux"
-)
-
-type autoInc struct {
-    sync.Mutex
-    id int
-}
-
-func (a *autoInc) ID() (id int) {
-    a.Lock()
-    defer a.Unlock()
-
-    id = a.id
-    a.id++
-    return
-}
-
-var unique autoInc
-
-type event struct {
-	ID          int		`json:"ID"`
-	Title       string	`json:"Title"`
-	Description string	`json:"Description"`
-}
-
-var events = []event{
-	{
-		ID:          unique.ID(),
-		Title:       "Introduction to Golang",
-		Description: "Come join us for a chance to learn how golang works and get to eventually try it out",
-	},
-}
-
-func createEvent(w http.ResponseWriter, r *http.Request) {
-	var newEvent event
-	newEvent.ID = unique.ID()
-
-	reqBody, err := ioutil.ReadAll(r.Body)
-    if err != nil {
-        http.Error(w, err.Error(), http.StatusBadRequest)
-        return
-    }
-	json.Unmarshal(reqBody, &newEvent)
-	events = append(events, newEvent)
-	w.WriteHeader(http.StatusCreated)
-
-	json.NewEncoder(w).Encode(newEvent)
-}
-
-func getOneEvent(w http.ResponseWriter, r *http.Request) {
-	eventIDParam := mux.Vars(r)["id"]
-	eventID, err := strconv.Atoi(eventIDParam)
-    if err != nil {
-        http.Error(w, err.Error(), http.StatusBadRequest)
-        return
-    }
-
-	for _, singleEvent := range events {
-		if singleEvent.ID == eventID {
-			json.NewEncoder(w).Encode(singleEvent)
-		}
-	}
-}
-
-func getAllEvents(w http.ResponseWriter, r *http.Request) {
-	json.NewEncoder(w).Encode(events)

```

Comment:
Encode returns an error. Unclear to me if it can error and also still write the response? Idk - maybe we should capture the error and return a 500 if it fails?